src/commands/distribute/groups/update.ts

```diff
@@ -58,6 +58,12 @@ export default class UpdateDistributionGroupCommand extends AppCommand {
   @hasArg
   public testersToDeleteListFile: string;
 
+  @help("Whether the distribution group is public (allowing anyone to download the releases)")
+  @shortName("p")
+  @longName("isPublic")
+  @hasArg
+  public isPublic: string;
```

Comment:
`public` or maybe `is-public` though the `is` is not relevant.
Do we natively support bool here by any chance?

Reply:
Also it looks like this should act like a switch, where `-p` (or even `--public`) alone should set the value to `true`.

Reply:
I considered that, but then how do you set it to not public? With another switch?

Reply:
`--public=false`

Reply:
we apparently have support for booleans in the base class, let's see what it can do, and whether we need to expand it for this (which would not be worth the effort)

Reply:
Alternatively, a common convention is to prefix with `no` as in `--no-public` (no short form here)

Reply:
We're using names like `isMandatory` and `isDisabled` in other places, but I can rename it to just `public`. We also have precedent for that. It's a mix.

This is not a flag, it's a setter. It's setting a value to true or false, not on the command but on the distribution group it's modifying.

Problem with making it bool is that it will need to default to false, and we'll no longer be able to check whether it is passed or not. TypeScript doesn't support nullable booleans. We can't work with that, because we can't just go set the value based on that information. Since it's a setter, not a flag.

If providing a string value for a boolean is really unwanted, we can have two parameters called `makePublic` and `makePrivate` or something. I'm not sure that looks nicer.

Reply:
> We're using names like `isMandatory` and `isDisabled` in other places, but I can rename it to just `public`. We also have precedent for that. It's a mix.

Sorry, but that's not true. From the code right there:
```
  @help("Specifies whether this release should be considered mandatory. (Putting -m flag means mandatory)")
  @shortName("m")
  @longName("mandatory")
  public isMandatory: boolean;
```

The option is called `mandatory` and it's a boolean switch. Specifically for casing, the only mistake I could find was `displayKeys` (we may want to consider fixing that in another PR, it should be `display-keys`).

> This is not a flag, it's a setter. It's setting a value to true or false, not on the command but on the distribution group it's modifying.

I understand that. There are ways to make this work. Either the option could also accept an optional argument, or we could use something like `--no-public` or `--private` as you proposed.

> Problem with making it bool is that it will need to default to false, and we'll no longer be able to check whether it is passed or not. TypeScript doesn't support nullable booleans. We can't work with that, because we can't just go set the value based on that information. Since it's a setter, not a flag.

The default value of a boolean field is "the field does not even exist", and "undefined" is perfectly acceptable for a boolean in TS. Yay.

The only question is how much effort is it to have an optional argument to a boolean switch, not whether it's possible or impossible.

Reply:
One more vote for having it as `boolean` using `undefined`. Otherwise we should handle the string to be a boolean value and that sound like more work.

Reply:
You are right on the naming. I was looking at the variable names, not the argument names.

The way booleans work in the CLI is by not setting the `hasArg` attribute on a property. It then becomes a boolean (regardless of what you say it should be in the property definition). The value is `false` when not provided. There is no `undefined`. Unless we add support for an optional parameter, which would be difficult*, there is mathematically speaking no difference between `false` and `undefined`. It's just the "not true" option, and cannot be used. So we need two switches.

* It would require defining an actual switch type parameter with a new syntax for taking an optional `false` argument.

There is no other example that I know of in the CLI that sets a boolean as obviously as this one, so we have to make a new decision. You and @yujincat are in favor of having two switches (or against a parameter taking a "true"/"false" string), and I don't have an opinion. So let's go with the two switches. I'll call them `--public` and `--no-public` as you suggested.

This is a good discussion btw, I'm glad we're having it :)
Things may be built on top of this in the future.